### ISSUE TITLE
MetricConfig for GPU SubCluster

### ISSUE THREAD

**Issue by giesselmann** [ClusterCockpit/cc-backend#99]
Hi,

I have a question/issue regarding the config of visible metrics for subclusters. The use-case is different slurm partitions, one with GPUs. Only the GPU nodes report GPU usage etc and only jobs running on these nodes should display the respective metrics. Metrics like load, mem_used etc. are consistent across partitions and would ideally only be configured once.

My cluster.json is structured as follows:
```
{
 "name": "levante",
   "subClusters": [
     {
       "name": "compute", ...
     },
     {
       "name": gpu",...
       "topology": {...
         "accelerators" : [...]
       }
     }
    ],
    "metricConfig": [
      {e.g. cpu_load},
      {e.g. mem_used},
}
```

If I would now add GPU-usage to the global metricConfig, this metric would on compute nodes not be available and in the job-view be rendered as warning, metric missing.

I've already seen, that the schema MetricConfig has a field subClusters, I couldn't figure out, what it is used for.
```
type MetricConfig {
  name:        String!
  unit:        String!
  scope:       MetricScope!
  aggregation: String
  timestep:    Int!
  peak:        Float
  normal:      Float
  caution:     Float
  alert:       Float
  subClusters: [SubClusterConfig]
}
```

What I basically need is a way to specify for which subClusters a metric is available. 
Or sth. like overloading metric configs, e.g. cpu_load is configured globally, but if for example different limits are needed for a subcluster, one can override the setting there.

And far far in the future, het-jobs will be a topic, thus jobs with nodes from different partitions, and in the CPU-GPU case with some metrics only from some nodes. This would currently also cause a list of missing metric warnings in the job-view. Worth to have it on the radar, but not urgent at all.

Not sure if special for us, or a general use case, let me know what you think.

**Comment by fodinabor** [comment]
Seconding, it'd be good to be able to select which metrics to show for which sub clusters.

Regarding setting different limits: that's exactly what the `subClusters` currently allows you to do. So a `disable` field in there would help, I guess?
```json
        {
            "name": "flops_any",
            "scope": "node",
            "unit": "GF/s",
            "timestep":60,"aggregation": "sum",
            "peak":9216,"normal":1000,"caution":200,"alert":50, // default values
            "subClusters": [
                {
                    "name": "P100-SXM2 E5-2690",
                    "peak":650,"normal":650,"caution":40,"alert": 20
                },
                {
                    "name": "P100-SXM2 E5-2698",
                    "peak":824,"normal":824,"caution":50,"alert": 20
                },
                {
                    "name": "P100-PCIE-16GB",
                    "peak":824,"normal":824,"caution":50,"alert": 20
                },
                {
                    "name": "Nvidia DGX V100",
                    "peak":824,"normal":824,"caution":50,"alert": 20
                },
                {
                    "name": "Epyc 7702",
                    "peak":1999,"normal":1999,"caution":100,"alert": 50
                },
                {
                    "name": "Supermicro A100",
                    "peak":1999,"normal":1999,"caution":100,"alert": 50
                },
                {
                    "name": "Supermicro A100 SXM",
                    "peak":1999,"normal":1999,"caution":100,"alert": 50
                }
            ]
        },
```

**Comment by giesselmann** [comment]
Nice, didn't know one could do that.
And yes, a disable field should do the job from my understanding.
And at some point additional logic could even handle het-jobs, displaying/masking metrics for parts of a job running on a different partition.

**Comment by moebiusband73** [comment]
The disable field is actually a simple and very good suggestion.

The thresholds are currently defined hierarchically. There is a default which can be overwritten for specific subclusters.
You could also do the same for the metric lists, have a default global list and add metrics to specific subclusters.

Another option would be to only take a metric into account for a subcluster if they appear in the SubClusterConfig for this subcluster, without the thresholds in case the global ones are valid.
To sum it up. In this case you would have a global metric list, where every metric has a list defining for which subclusters the metric is taken into account.

What do you think?

**Comment by moebiusband73** [comment]
Another thought: The suggested disable flag is of course much better in cases where most of the metrics are shared among subclusters. And this is probably the common case.

**Comment by fodinabor** [comment]
I like the `remove` approach, as in the example.

**Comment by giesselmann** [comment]
+1 remove or hide
